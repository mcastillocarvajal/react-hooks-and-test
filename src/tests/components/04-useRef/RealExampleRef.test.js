import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Test for <RealExampleRef />', () => {
    
    const wrapper = shallow( <RealExampleRef /> );

    test('should show component correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show component <MultipleCustomHooks />', () => {
        
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
    })
    
})
