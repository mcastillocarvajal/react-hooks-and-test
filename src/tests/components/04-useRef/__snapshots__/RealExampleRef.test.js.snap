// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <RealExampleRef /> should show component correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-dark mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
